kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -0,0 +1,307 @@
+                deploymentService = (ListenerSupport) service;
+            } else if (processService == null && ProcessService.class.isAssignableFrom(service.getClass())) {
+                processService = (ProcessService) service;
+            }
+            if (deploymentService != null && processService != null) {
+                break;
+            }
+        }
+        if (deploymentService == null) {
+            throw new IllegalStateException("Cannot find deployment service");
+        }
+        if (processService == null) {
+            throw new IllegalStateException("Cannot find process service");
+        }
+        deploymentService.addListener(this);
+    }
+
+    @Override
+    public void destroy(KieServerImpl kieServer, KieServerRegistry registry) {
+        pollService.shutdownNow();
+        pollService = null;
+        notifyService.shutdownNow();
+        notifyService = null;
+        if (deploymentService != null) {
+            deploymentService.removeListener(this);
+            deploymentService = null;
+        }
+        if (consumer != null) {
+            consumer.close();
+            consumer = null;
+        }
+        initialized.set(false);
+    }
+
+    protected Consumer<String, byte[]> getKafkaConsumer() {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, System.getProperty(
+                KAFKA_EXTENSION_PREFIX + "boopstrap.servers", "localhost:9092"));
+        // read only committed events
+        props.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, IsolationLevel.READ_COMMITTED.toString().toLowerCase());
+        // do not automatically create topics by default
+        props.put(ConsumerConfig.ALLOW_AUTO_CREATE_TOPICS_CONFIG, Boolean.getBoolean(
+                KAFKA_EXTENSION_PREFIX + "auto.create.topics"));
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, System.getProperty(KAFKA_EXTENSION_PREFIX + "group.id",
+                "jbpm-consumer"));
+        return new KafkaConsumer<>(props, new StringDeserializer(), new ByteArrayDeserializer());
+    }
+
+    @Override
+    public void onDeploy(DeploymentEvent event) {
+        updateRegistration(event, this::updateTopics);
+    }
+
+    @Override
+    public void onUnDeploy(DeploymentEvent event) {
+        updateRegistration(event, this::removeTopics);
+    }
+
+    @Override
+    public void onActivate(DeploymentEvent event) {
+        updateRegistration(event, this::updateTopics);
+    }
+
+    @Override
+    public void onDeactivate(DeploymentEvent event) {
+        updateRegistration(event, this::removeTopics);
+    }
+
+    private void updateTopics(String deploymentId, ProcessDefinition processDefinition) {
+        for (SignalDesc signal : processDefinition.getSignalsMetadata()) {
+            topic2Signal.computeIfAbsent(topicFromSignal(signal), t -> new SignalInfo(deploymentId,
```

Comment:
if we have several deployments with the same signal name, wouldn't be overriding the deployment Id ?

Reply:
Map structure changed to fix this